src/main/java/com/github/splendor_mobile_game/game/model/TokenList.java

```diff
@@ -8,17 +8,71 @@ public class TokenList {
 
     private ArrayList<Token> tokens;
 
-    private TokenType tokenType;
+    private final TokenType tokensType;
 
-    public TokenList(TokenType tokenType) {
-        this.tokenType = tokenType;
+    private int availableTokens = 7;
+
+    public TokenList(TokenType tokensType) {
+        this.tokensType = tokensType;
     }
 
-    public boolean addToken(Token token) {
-        if (tokenType != token.getTokenType()) return false;
+    public void addToken(Token token) {
+        if (tokensType != token.getTokenType()) return;
 
         this.tokens.add(token);
-        return true;
     }
 
+
+
+
+    public int getAvailableTokensCount() {
+        return availableTokens;
+    }
+
+
+    public int getEquippedTokensCount() {
+        return 7 - availableTokens;
+    }
+
+    /**
+     * Assign a token to user
+     *
+     * @param user -> player which will be the owner of available token.
+     * @return Token object, including its uuid
+     */
+    public Token equipToken(User user) {
+        for (Token token : tokens) {
+            if (token.getOwner() == null) {
+                token.setOwner(user);
+                availableTokens--;
+            }
+        }
+        return null;
+    }
+
+
+    /**
+     * Dissociate a token from user
+     *
+     * @param user -> player which we are dissociating token from
+     * @return Token object, including its uuid
+     */
+    public Token unequipToken(User user) {
+        for (Token token : tokens) {
+            if (token.getOwner().equals(user)) {
+                token.setOwner(null);
+                availableTokens++;
+            }
+        }
+        return null;
+    }
```

Comment:
First I thought this would be very inefficient, but taking into consideration our assumptions about game mechanics this is an O(1) solution. 

“Premature optimization is the root of all evil” - Donald Knuth